gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -569,6 +569,18 @@ public boolean handle(final RoutingContext ctx) {
             }
         }
 
+        HttpServerResponse response = ctx.response();
+        String queryParam = null;
+
+        if (request.params() != null) {
+            queryParam = request.getParam("q");
+        }
+
+        if (queryParam != null && !queryParam.isEmpty()) {
+            this.handleHookSearch(queryParam, response);
```

Comment:
The `handleHookSearch` should be executed for `GET` requests to `/registrations/listeners` or `/registrations/routes` only. A request to the hookUrl will trigger the same code and there we don't want to search for hooks

Reply:
Ok I will validate this.